test/test_broker.hpp

```diff
@@ -963,10 +973,20 @@ class test_broker {
 
         MQTT_NS::buffer client_id;
         MQTT_NS::optional<MQTT_NS::will> will;
-        bool session_clear = !act_sess_it->session_expiry_interval;
-        if (ep.clean_session() && session_clear) {
-            client_id = std::move(act_sess_it->client_id);
-            will = std::move(act_sess_it->will);
+        bool session_clear =
+            [&] {
+                if (ep.get_protocol_version() == MQTT_NS::protocol_version::v3_1_1) {
+                    return ep.clean_session();
+                }
+                else {
+                    BOOST_ASSERT(ep.get_protocol_version() == MQTT_NS::protocol_version::v5);
+                    auto const& sei_opt = (*act_sess_it)->session_expiry_interval;
+                    return !sei_opt || sei_opt.value() == std::chrono::steady_clock::duration::zero();
```

Comment:
I think 

```cpp
std::chrono::seconds::zero()
```

instead?

Reply:
Yes, you are right! I will fix it.

Reply:
I noticed that there are many place using `std::chrono::steady_clock::duration::zero()`.
For example 
https://github.com/redboltz/mqtt_cpp/blob/5ae194f8a43d8b910d9d4540477d29d862f9c808/include/mqtt/client.hpp#L369

I'm not 100% sure but it is introduced by you. (not checked log...)

The type of `sessions::session_expiry_interval` is `MQTT_NS::optional<std::chrono::steady_clock::duration> session_expiry_interval`.

I think that `std::chrono::steady_clock::duration::zero()` is more natural choice.
What do you think?

Reply:
This would have been me, yes.

Hrmm.

I do not know which is best.

See https://en.cppreference.com/w/cpp/chrono/steady_clock

steady_clock::duration::zero() is not necessarily seconds::zero().

Maybe better to use steady_clock?

Reply:
I think so.
I keep `std::chrono::steady_clock::duration::zero()` code.